libs/SalesforceReact/src/com/salesforce/androidsdk/reactnative/bridge/SmartSyncReactBridge.java

```diff
@@ -250,7 +250,10 @@ private void handleSyncUpdate(final SyncState sync, Callback successCallback, Ca
                     ReactBridgeHelper.invokeSuccess(successCallback, sync.asJSON());
                     break;
                 case FAILED:
-                    errorCallback.invoke("Sync failed");
+                    //errorCallback.invoke("Sync failed");
+
+                    //Return sync to React Native with the error message in the JSON
+                    ReactBridgeHelper.invokeSuccess(errorCallback, sync.asJSON());
```

Comment:
Could you rename ReactBridgeHelper.invokeSuccess methods to ReactBridgeHelper.invoke and rename also the first argument to those methods from successCallback to callback?

Reply:
Resolved in c2d0a02bb179ceda2f52773f03a13e4e19fed34f